To: vim_dev@googlegroups.com
Subject: Patch 7.3.430
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.430
Problem:    When a custom filetype detection uses "augroup END" the conf
            fileytpe detection does not have the filetypedetect group.
Solution:   Always end the group and include filetypedetect in the conf
            autocommand. (Lech Lorens)
Files:      runtime/filetype.vim
    

*** ../vim-7.3.429/runtime/filetype.vim	2012-02-05 20:08:08.000000000 +0100
--- runtime/filetype.vim	2012-02-05 19:59:40.000000000 +0100
***************
*** 2545,2561 ****
  " detected filetypes.
  runtime! ftdetect/*.vim
  
  
  " Generic configuration file (check this last, it's just guessing!)
! au BufNewFile,BufRead,StdinReadPost *
  	\ if !did_filetype() && expand("<amatch>") !~ g:ft_ignore_pat
  	\    && (getline(1) =~ '^#' || getline(2) =~ '^#' || getline(3) =~ '^#'
  	\	|| getline(4) =~ '^#' || getline(5) =~ '^#') |
  	\   setf conf |
  	\ endif
  
- augroup END
- 
  
  " If the GUI is already running, may still need to install the Syntax menu.
  " Don't do it when the 'M' flag is included in 'guioptions'.
--- 2545,2563 ----
  " detected filetypes.
  runtime! ftdetect/*.vim
  
+ " NOTE: The above command could have ended the filetypedetect autocmd group
+ " and started another one. Let's make sure it has ended to get to a consistant
+ " state.
+ augroup END
  
  " Generic configuration file (check this last, it's just guessing!)
! au filetypedetect BufNewFile,BufRead,StdinReadPost *
  	\ if !did_filetype() && expand("<amatch>") !~ g:ft_ignore_pat
  	\    && (getline(1) =~ '^#' || getline(2) =~ '^#' || getline(3) =~ '^#'
  	\	|| getline(4) =~ '^#' || getline(5) =~ '^#') |
  	\   setf conf |
  	\ endif
  
  
  " If the GUI is already running, may still need to install the Syntax menu.
  " Don't do it when the 'M' flag is included in 'guioptions'.
*** ../vim-7.3.429/src/version.c	2012-02-05 01:18:41.000000000 +0100
--- src/version.c	2012-02-05 20:02:18.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     430,
  /**/

-- 
The technology involved in making anything invisible is so infinitely
complex that nine hundred and ninety-nine billion, nine hundred and
ninety-nine million, nine hundred and ninety-nine thousand, nine hundred
and ninety-nine times out of a trillion it is much simpler and more
effective just to take the thing away and do without it.
		-- Douglas Adams, "The Hitchhiker's Guide to the Galaxy"

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
